parser_test.go

```diff
@@ -1498,10 +1498,10 @@ func TestArrayEachEmpty(t *testing.T) {
 	}{
 		{"Empty array", args{[]byte("[]"), funcError, []string{}}, 1, false},
 		{"Empty array with space", args{[]byte("[ ]"), funcError, []string{}}, 2, false},
-		{"Empty array with \n", args{[]byte("[\n]"), funcError, []string{}}, 2, false},
+		{"Empty array with newline", args{[]byte("[\n]"), funcError, []string{}}, 2, false},
 		{"Empty field array", args{[]byte("{\"data\": []}"), funcError, []string{"data"}}, 10, false},
 		{"Empty field array with space", args{[]byte("{\"data\": [ ]}"), funcError, []string{"data"}}, 11, false},
-		{"Empty field array with \n", args{[]byte("{\"data\": [\n]}"), funcError, []string{"data"}}, 11, false},
+		{"Empty field array with newline", args{[]byte("{\"data\": [\n]}"), funcError, []string{"data"}}, 11, false},
```

Comment:
do we need a test for array with an empty string, ie `[]byte("{\"data\": [\"\"]}"

Reply:
As I understood, this is just a test to iterate over an array, not accessing an element by index in that array. The API of this library is rather confusing tbh.